### ISSUE TITLE
Release branch is too divergent from 'master'

### ISSUE THREAD

**Issue by tseaver** [googleapis/releasetool#66]
For instance, I just ran `releasetool` to create https://github.com/googleapis/google-cloud-python/pull/6139.  The diff *shown* in the PR looks right, but the following causes CI to [trigger for inappropriate sections](https://circleci.com/gh/googleapis/google-cloud-python/8506?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):

```bash
$ git checkout master && git fetch --all --prune &&  git merge upstream/master
Switched to branch 'master'
Your branch is up-to-date with 'upstream/master'.
Fetching origin
Fetching upstream
Already up-to-date.
.circleci/config.yml
.kokoro/build.sh
.kokoro/continuous/asset.cfg
.kokoro/continuous/automl.cfg
....
kokoro/continuous/translate.cfg
.kokoro/continuous/websecurityscanner.cfg
.kokoro/presubmit/asset.cfg
.kokoro/presubmit/automl.cfg
...
.kokoro/presubmit/translate.cfg
.kokoro/presubmit/websecurityscanner.cfg
README.rst
api_core/CHANGELOG.md
api_core/nox.py
api_core/setup.cfg
api_core/setup.py
bigquery/google/cloud/bigquery/client.py
bigquery/google/cloud/bigquery/job.py
bigquery/tests/unit/test_client.py
bigquery/tests/unit/test_job.py
bigtable/README.rst
bigtable/setup.py
dlp/CHANGELOG.md
dlp/setup.py
dlp/synth.py
firestore/tests/unit/testdata/create-all-transforms.textproto
firestore/tests/unit/testdata/create-arrayremove-multi.textproto
...
firestore/tests/unit/testdata/update-paths-st-with-empty-map.textproto
firestore/tests/unit/testdata/update-st-with-empty-map.textproto
logging/google/cloud/logging/_helpers.py
logging/google/cloud/logging/client.py
logging/google/cloud/logging/entries.py
logging/google/cloud/logging/handlers/app_engine.py
logging/google/cloud/logging/logger.py
logging/tests/unit/handlers/test_app_engine.py
logging/tests/unit/test__helpers.py
logging/tests/unit/test_entries.py
logging/tests/unit/test_logger.py
pubsub/setup.py
storage/setup.py
storage/tests/system.py
storage/tests/unit/test_blob.py
storage/tests/unit/test_bucket.py
trace/README.rst
```

When I ran `releasetool`, I was on an updated version of the master branch.

**Comment by theacodes** [comment]
I *think* this is just a problem with the logic we use in circleci to compute the differences, because I changed that quite a bit for Kokoro. WDYT, @tseaver?

**Comment by tseaver** [comment]
@tseaver I'd be willing to let it go, but the `git diff master..HEAD` output tells me there is something wrong with the way releasetool is making the branch:  it should be rooted from the current `HEAD` of master, with only the changes it generates on top.  Instead, it looks as though it is rooted against an older revision of `master` (maybe the last one which touched the API subdir?)

**Comment by theacodes** [comment]
Hrm, weird. Let's definitely look into it.

**Comment by busunkim96** [comment]
I did some investigating. I believe this is because `origin/master` is used as the base branch when the new release branch is made (rather than `master`). 

https://github.com/googleapis/releasetool/blob/37e2967997a953d119aa18c91a45c58157dc4ebc/releasetool/commands/start/python.py#L127-L130
https://github.com/googleapis/releasetool/blob/37e2967997a953d119aa18c91a45c58157dc4ebc/releasetool/git.py#L40-L41

I was able to recreate this situation by running `releasetool start` with an up-to-date `master` and a slightly older fork.
